### DISCUSSION TITLE
Cannot make ADFS work. Need help

### DISCUSSION THREAD

**Discussion by hchautrung** [bergie/passport-saml#935] [Q&A]
Hi all,

I got issue with ADFS to make it work. Please advise I need to make it work asap.

The administrator update  the link of ADFS Metadata URL https://fs.singaporetech.edu.sg/federationmetadata/2007-06/federationmetadata.xml and configuration in the server below: 

![image](https://github.com/user-attachments/assets/1f35a3ba-d2ab-4497-9e7f-317e220b1759)
![image](https://github.com/user-attachments/assets/5041bac6-6822-40b7-9aad-be68b931168c)
![image](https://github.com/user-attachments/assets/91d6e385-bcbb-400a-a41a-c984de57d7c4)  

I used `@node-saml/passport-saml, passport-saml-metadata` and `passport` packages to implement SSO in Node.js. Here is the test `server.js` and `config.js` that I follow the instruction from [Usage with Active Directory Federation Services](https://github.com/node-saml/passport-saml)  
![image](https://github.com/user-attachments/assets/8d5be7b3-2146-4691-b5c2-3f4fdf84d44a)

1. **`Server.js`**  

![image](https://github.com/user-attachments/assets/022aa58a-7ea5-4d46-8401-5d3b30bf57ab)

2. **`config/passport.js`**   

![image](https://github.com/user-attachments/assets/6372e96c-0f50-4164-b378-6b0f75a8c761)  

The `federationmetadata.xml` file is provided by ADFS server via link [ttps://fs.singaporetech.edu.sg/federationmetadata/2007-06/federationmetadata.xml](ttps://fs.singaporetech.edu.sg/federationmetadata/2007-06/federationmetadata.xml)

Backend node.js log to check SamlStratergy  
![image](https://github.com/user-attachments/assets/b1aae87a-37c5-4ba2-84b4-a3b88dcdefad)

Access the url: [https://llos.unimetaverse.net/](https://llos.unimetaverse.net/) and click login and got error

**Comment by srd90** [comment]
Your example codes are screenshots. While there are good OCR tools to get text out of those it is annoying thing to do.

You could have provided more information about this "..and got error". What error? Any stacktrace available?

passport-saml version is also missing from the question.

It seems that at least your SSO callback is not implemented correctly. You should delegate handling of authentication response to passport-saml. I bet you have configured authnrequest to be delivered via HTTP POST binding. Following code is copy pasted from one of the images with OCR tool and incorrectly recognized parts of the text manually fixed (unneccessary extra work). This code does not delegate to passport-saml.

>```javascript
>app.post('/sso/callback', function(req, res, next) {
>    console.log('/sso/callback POST');
>    res.send('/sso/callback POST');
>});
>```

But since you didn't provide any hint of the error there is a chance that you haven't even got this far due e.g. some error caused by authentication request sent to IdP.

FWIW, once SAML 2.0 authn response flows to passport-saml component there is a chance (depending on how your ADFS is configured and how your stack is configured) that you shall get error that signature is missing...but you are not that far yet. Once you are and if you encounter such error read through issues/links at this comment https://github.com/node-saml/node-saml/issues/378#issuecomment-2344755012 furthermore read these https://github.com/node-saml/passport-saml/issues/816 and https://github.com/node-saml/passport-saml/pull/840#issuecomment-1397272720
